tests/Unit/ControlSystem/Test_Measurements.cpp

```diff
@@ -23,3 +34,136 @@ static_assert(
+      CHECK(2 == deriv_order);
+    }
+  };
+};
+
+template <typename Metavariables>
+struct MockControlSystemComponent {
+  using component_being_mocked =
+      ControlComponent<Metavariables, MockControlSystem>;
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockSingletonChare;
+  using array_index = int;
+  using simple_tags_from_options = tmpl::list<>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<Parallel::Phase::Testing, tmpl::list<>>>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<MockControlSystemComponent<Metavariables>>;
+};
+
+}  // namespace
+
+// This test tests the calculation of the center of mass of a star
+// in the FindTwoCenters submeasurement
+SPECTRE_TEST_CASE("Unit.ControlSystem.FindTwoCenters",
+                  "[ControlSystem][Unit]") {
+  // Part 1 of the test: calculation of the relevant integrals
+  // within a (mock) element.
+  const Mesh<3> mesh(2, Spectral::Basis::FiniteDifference,
+                     Spectral::Quadrature::CellCentered);
+  const Scalar<DataVector> tilde_d{
+      DataVector{0.0, 0.5, 1.0, 1.0, 0.5, 0.0, 1.0, 1.0}};
+  const Scalar<DataVector> inv_det_jacobian{
+      DataVector{0.2, 0.2, 0.4, 0.4, 0.5, 0.5, 1.0, 1.0}};
+  const DataVector x_coord{-1.0, 1.0, -1.0, 1.0, 1.0, 2.0, 0.0, 2.0};
+  const DataVector y_coord{0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0};
+  const DataVector z_coord{0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0};
+  const tnsr::I<DataVector, 3, Frame::Grid> x_grid{{x_coord, y_coord, z_coord}};
+
+  // mass_a and mass_b: Integral of tilde_d/inv_det_jacobian for x>=0 and x<0
+  // first_moment_a, first_moment_b : Same as masses, but multiplying the
+  //                              integrand by x_grid
+  double mass_a = 0.;
+  double mass_b = 0.;
+  std::array<double, 3> first_moment_a = {0., 0., 0.};
+  std::array<double, 3> first_moment_b = {0., 0., 0.};
+  control_system::measurements::center_of_mass_integral_on_element(
+      &mass_a, &mass_b, &first_moment_a, &first_moment_b, mesh,
+      inv_det_jacobian, tilde_d, x_grid);
+  // Comparison with expected answer
+  CHECK(mass_a == 8.0);
+  CHECK(mass_b == 2.5);
+  CHECK(first_moment_a == std::array<double, 3>{8.0, 4.5, 3.0});
+  CHECK(first_moment_b == std::array<double, 3>{-2.5, 2.5, 0.0});
+
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+  using control_system_component = MockControlSystemComponent<Metavariables>;
+
+  MockRuntimeSystem runner{{}};
+  ActionTesting::emplace_singleton_component<control_system_component>(
+      make_not_null(&runner), ActionTesting::NodeId{0},
+      ActionTesting::LocalCoreId{0});
+  ActionTesting::set_phase(make_not_null(&runner), Parallel::Phase::Testing);
+  auto& cache = ActionTesting::cache<control_system_component>(runner, 0);
+
+  LinkedMessageId<double> measurement_id;
+  measurement_id.id = 1.0;
+  measurement_id.previous = 0.0;
```

Comment:
This can be initialized as `LinkedMessageId<double> measurement_id{1.0, 0.0};`

Reply:
Done